romanesco/plugins/swift/executor.py

```diff
@@ -0,0 +1,78 @@
+import os
+import re
+import romanesco.utils
+import tempfile
+
+
+def _transform_path(inputs, taskInputs, inputId, tmpDir):
+    """
+    If the input specified by inputId is a filepath target, we transform it to
+    its absolute path within the docker container (underneath /data).
+    """
+    for ti in taskInputs.itervalues():
+        tiId = ti['id'] if 'id' in ti else ti['name']
+        if tiId == inputId:
+            if ti.get('target') == 'filepath':
+                rel = os.path.relpath(inputs[inputId]['script_data'], tmpDir)
+                return os.path.join('/data', rel)
+            else:
+                return inputs[inputId]['script_data']
+
+    raise Exception('No task input found with id = ' + inputId)
+
+
+def _expand_args(args, inputs, taskInputs, tmpDir):
```

Comment:
Since this calls `_transform_path` I'm tempted to leave this in both places for now - the swift one will not call `_transform_path` but the docker one will. I'll remove `_transform_path` from the swift plugin. Otherwise either (1) a transform function would need to be passed in (identity function for swift) or (2) path transforms would be a pre- or post-processing step, not embedded in `_expand_args`. I lean toward (2) as a pre-processing step, but don't feel like it's worth messing with at this point.